commitment/tap.go

```diff
@@ -120,17 +145,37 @@ func NewTapCommitment(newCommitments ...*AssetCommitment) (*TapCommitment,
 		return nil, err
 	}
 
-	return &TapCommitment{
-		Version:          maxVersion,
-		TreeRoot:         root,
-		assetCommitments: assetCommitments,
-		tree:             tree,
-	}, nil
+	switch {
+	// If no version was specified, we'll use the max version of the asset
+	// commitments.
+	case version == nil:
+		return &TapCommitment{
+			TreeRoot:         root,
+			assetCommitments: assetCommitments,
+			tree:             tree,
+			Version:          TapCommitmentVersion(maxVersion),
```

Comment:
👍 for bubbling this routine up, and also making it more explicit.

Reply:
Ah the last case could be dropped here actually.